gpii/node_modules/packagekit/nodepackagekit/nodepackagekit.cc

```diff
@@ -29,13 +37,20 @@ static gboolean DEBUG = FALSE;
 
 static gchar TAG[] = "[nodePackagekit]";
 
+typedef struct {
+  const gchar* property;
+  const gchar* (*accessFunc)(PkPackage *package);
+} PkitPropAccess;
+
 static std::basic_string<gchar>
-utf8StringFromValue(v8::Handle<v8::Value> value)
+utf8StringFromValue (Handle<Value> value)
 {
-    v8::Local<v8::String> string = value->ToString();
-    gchar *buffer = g_new(gchar, string->Utf8Length());
-    string->WriteUtf8(buffer);
-    return buffer;
+    std::basic_string<gchar> result;
+    Nan::Utf8String nanUtf8 (value);
+    if (nanUtf8.length() != 0) {
```

Comment:
According to the [Nan documentation](https://github.com/nodejs/nan/blob/master/doc/v8_misc.md#nanutf8string), if the Nan::Utf8String fails to be created, then length() returns zero, but using the * operator (used in the next line) gives a NULL result and not an empty string.  Thus, if length() is zero, there's nothing to append, and don't bother trying.

Unless I'm reading the Nan docs wrong.